collectors/bundles/bundles.go

```diff
@@ -52,6 +58,40 @@ func GetFinalizedBundle(restEndpoint string, poolId int64, bundleId int64) (*typ
 	return &finalizedBundle, nil
 }
 
+func GetFinalizedBundleByIndex(restEndpoint string, poolId int64, index int64) (*types.FinalizedBundle, error) {
+	raw, err := utils.GetFromUrlWithBackoff(fmt.Sprintf(
+		"%s/kyve/v1/bundles/%d?index=%d",
+		restEndpoint,
+		poolId,
+		index,
+	))
+	if err != nil {
+		return nil, err
+	}
+
+	var bundlesResponse types.FinalizedBundlesResponse
+
+	if err := json.Unmarshal(raw, &bundlesResponse); err != nil {
+		return nil, err
+	}
+
+	if len(bundlesResponse.FinalizedBundles) == 1 {
+		return &bundlesResponse.FinalizedBundles[0], nil
+	}
+
+	return nil, fmt.Errorf("failed to find finalized bundle for index %d: %w", index, err)
+}
+
+func GetFinalizedBundleForBlockHeight(chainRest string, blockPool types.PoolResponse, height int64) (*types.FinalizedBundle, error) {
+	startKey, err := strconv.ParseInt(blockPool.Pool.Data.StartKey, 10, 64)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse start key %s: %w", blockPool.Pool.Data.StartKey, err)
+	}
+
+	// index is height - startKey
```

Comment:
// index is height - startKey is correct here since we have our targetHeight and now we need to find the index of the height so we can directly request the right bundle with it